charts/pega/templates/_pega-eks-ingress.tpl

```diff
@@ -66,7 +73,16 @@ spec:
 {{ include "ingressServiceC11n" . | indent 10 }}
       {{ end }}
       - pathType: ImplementationSpecific
-        backend: 
-{{ include "ingressService" . | indent 10 }}
+        backend:
+# protocol will be set to https only when either ingress is enabled or domain is set
+{{- if or (.node.service.domain) (.node.ingress) }}
+{{- if (.node.tls).enabled }}
+    {{ include "ingressServiceHttps" . | indent 10 }}
+{{- else }}
+    {{ include "ingressService" . | indent 10 }}
+{{- end }}
+{{- else }}
+    {{ include "ingressService" . | indent 10 }}
+{{- end }}
```

Comment:
Above few lines of code are common across all ingress templates. Would it be better to extract to helper template?

Reply:
done